server/routes/journeys/update-review/controller.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response } from 'express'
+import type PrisonerSearchService from '../../../services/prisonerSearch/prisonerSearchService'
+import { BaseJourneyController } from '../base/controller'
+import CsipApiService from '../../../services/csipApi/csipApiService'
+import { getNonUndefinedProp } from '../../../utils/utils'
+import { attendeeSorter } from '../../../utils/sorters'
+
+export class UpdateReviewController extends BaseJourneyController {
+  constructor(
+    override readonly csipApiService: CsipApiService,
+    private readonly prisonerSearchService: PrisonerSearchService,
+  ) {
+    super(csipApiService)
+  }
+
+  UPDATE = async (req: Request, res: Response) => {
+    const record = req.journeyData.csipRecord!
+    if (!record.plan?.reviews?.length) {
+      return res.redirect(`/csip-records/${record.recordUuid}`)
+    }
+
+    req.journeyData.prisoner = await this.prisonerSearchService.getPrisonerDetails(req, record.prisonNumber)
+    const review = record.plan.reviews.reduce(
+      (prev, cur) => (prev!.reviewSequence > cur.reviewSequence ? prev : cur),
+      record.plan.reviews[0],
+    )!
+
+    req.journeyData.review = {
+      reviewUuid: review.reviewUuid,
+      ...getNonUndefinedProp(review, 'reviewDate'),
+      ...getNonUndefinedProp(review, 'nextReviewDate'),
+      ...getNonUndefinedProp(review, 'summary'),
+      outcome: review.csipClosedDate ? 'CLOSE_CSIP' : 'REMAIN_ON_CSIP',
```

Comment:
This will "fix" the bad NOMIS data removing any other actions they'd set as well as the ones where both remain and close are set won't it? That seems fine to me especially as it's audited. Have we checked with the team that it's what was intended?

Reply:
yes. cfm'd with Theo it's okay to use csipClosedDate.
the alternative logic i suggested was to check whether `actions` contain `CLOSE_CSIP`, but checking csipCLosedDate is more straightforward. and Anthony has confirmed that all prod reviews with `CLOSE_CSIP` also have a closedDate.